server/common/data_locator.py

```diff
@@ -125,3 +128,25 @@ def __enter__(self):
     def __exit__(self, *args):
         if self.delete:
             os.unlink(self.tmp_path)
+
+
+def discover_region_name(uri):
```

Comment:
Will this work on non-AWS S3-compatible object stores or is it hard-wired to AWS only?

Ideally, it would gracefully degrade to non-AWS S3 stores that may not have a region.

Reply:
I suspect the code as written would throw and exception if it could not discover a region name.  I'll change this function to return something else if a region is not found, and then handle errors in the eb/app